File: q1_two_sum/two_sum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target) {
		
	int i, j;
	int c;

	int *result = (int *)malloc(sizeof(int)*2);

	for (i = 0; i < numsSize; ++i){

		c = target - nums[i];

		for (j = 0; j < numsSize; ++j){

			if (c == nums[j] && j != i){
				
				result[0] = i;
				result[1] = j;
				return result;
			}
			
		}
	
	}

	return ;

}

File: q1_two_sum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m = {}
        l = len(nums)

        # 48ms
        for i in range(0, l):
            c = target - nums[i]
            if m.has_key(c):
                if m[c] != i:
                    return [i, m[c]]
            m[nums[i]] = i
            
        '''
        # 60ms
        for i in range(0, l):
        	m[nums[i]] = i
        for i in range(0, l):
        	t = target - nums[i]
        	if m.has_key(t):
        	    if i != m[t]:
        		    return [i, m[t]]
        '''
